### PR TITLE
ci: Increase curl connect timeout to 15 seconds

### PR DESCRIPTION
**Pull request by gandro**
This increases the curl connection timeout from 5 to 15 seconds to avoid
issues with IPCache propagation delay. On Cilium master and 1.10, it
seems that IPCache updates in CI can take up to 4-8 seconds.

CI flakes potentially caused by the increased IPCache propagation delay:

 - #13839
 - #14959
 - #15103
 - #16237

### CODE REVIEW

**Review by tklauser — APPROVED** [review]
Thanks!

**Comment by gandro** [comment]
test-me-please

File: test/helpers/cons.go [hunk 1]
```diff
@@ -156,7 +156,9 @@ const (
 
 	// CurlConnectTimeout is the timeout for the connect() call that curl
 	// invokes
-	CurlConnectTimeout = 5
+	// FIXME: Revert this back to 5 seconds once the IPCache propagation delay
+	// (which seems to take up to 4-8 seconds) has been addressed.
```
Comment (nbusseneau) [inline test/helpers/cons.go]: [Nit] Can we add a link to a tracking issue?

**Review by nbusseneau — APPROVED** [review]

File: test/helpers/cons.go (hunk 1 above)
Comment (gandro) [reply test/helpers/cons.go]: Done!

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
GitHub Actions seems to have a really bad day. Had to reopen to re-trigger the workflows.

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/5555/

The GKE failure unfortunately looks legit and introduces by this approach. Because it tries longer to connect, we will see more packets getting denied. We will likely need to update the timeout on a per-test basis.

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
test-1.21-4.9

**Comment by gandro** [comment]
If https://github.com/cilium/cilium/pull/16381 passes CI soon, I think it might be better to merge it instead of this one, as it actually addresses the root cause and will fix all tests, not just the ones using curl.

**Comment by gandro** [comment]
Closing in favor of #16381

**Pull request closed without merging**